apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -763,6 +875,62 @@ static void utc_audio_pcm_writei_n(void)
 	TC_SUCCESS_RESULT();
 }
 
+
+/**
+* @testcase         audio_pcm_drain_p
+* @brief            play/record all enqueued buffers and stop the device
+* @scenario         play recored file previously completely
+* @apicovered       pcm_drain
+* @precondition     audio_pcm_writei_p should be executed just before this function
+* @postcondition    NA
+*/
+
+static void utc_audio_pcm_drain_p(void)
+{
+	/* Executed after writei positive tc */
+
+	int ret;
+
+	printf("Draining buffers to complete playback and close device\n");
+	ret = pcm_drain(g_pcm);
+	TC_ASSERT_GEQ_CLEANUP("pcm_drain", ret, 0, clean_all_data(0, NULL));
+	printf("Playback done!\n");
+
+	clean_all_data(0, NULL);
```

Comment:
Why do you call it for 0, NULL?

Reply:
We need to close the pcm device.  In this tc no file or buffer is allocated, so clean_all_data is called with 0 and NULL.